rpc/src/rpc.rs

```diff
@@ -507,6 +537,54 @@ impl JsonRpcRequestProcessor {
         })
     }
 
+    pub fn get_secondary_index_key_size(
+        &self,
+        index_key: &Pubkey,
+        commitment: Option<CommitmentConfig>,
+    ) -> Result<RpcResponse<HashMap<RpcAccountIndex, usize>>> {
+        // Acquire the bank
+        let bank = self.bank(commitment);
+
+        // Grab RPC server's secondary index config
+        // Exit if secondary indexes are not enabled
+        let secondary_index_config = if !self.config.account_indexes.is_empty() {
+            &self.config.account_indexes
+        } else {
+            debug!("get_secondary_index_key_size: secondary index not enabled.");
+            return Ok(new_response(&bank, HashMap::new()));
+        };
+
+        // Make sure your key is not explicitly excluded
+        if !self.config.account_indexes.include_key(index_key) {
+            return None.ok_or_else(|| {
+                Error::from(RpcCustomError::KeyExcludedFromSecondaryIndex {
+                    index_key: index_key.to_string(),
+                })
+            });
```

Comment:
Oh good catch, had custom errors in before and pulled em out in a refactor. Must have missed this.

Reply:
Humm actually getting this, not sure I can change it:
 ```error[E0277]: the trait bound `std::result::Result<solana_rpc_client_api::response::Response<std::collections::HashMap<solana_rpc_client_api::config::RpcAccountIndex, usize>>, jsonrpc_core::Error>: std::convert::From<solana_rpc_client_api::custom_error::RpcCustomError>` is not satisfied
   --> rpc/src/rpc.rs:564:15
    |
564 |             }.into();
    |               ^^^^ the trait `std::convert::From<solana_rpc_client_api::custom_error::RpcCustomError>` is not implemented for `std::result::Result<solana_rpc_client_api::response::Response<std::collections::HashMap<solana_rpc_client_api::config::RpcAccountIndex, usize>>, jsonrpc_core::Error>`
    |
    = help: the following other types implement trait `std::convert::From<T>`:
              <std::result::Result<(), idna::uts46::Errors> as std::convert::From<idna::uts46::Errors>>
              <std::result::Result<(), ring::error::Unspecified> as std::convert::From<ring::bssl::Result>>
              <std::result::Result<T, E> as std::convert::From<solana_rbpf::vm::StableResult<T, E>>>
              <std::result::Result<miniz_oxide::MZStatus, miniz_oxide::MZError> as std::convert::From<&miniz_oxide::StreamResult>>
              <std::result::Result<miniz_oxide::MZStatus, miniz_oxide::MZError> as std::convert::From<miniz_oxide::StreamResult>>
              <std::result::Result<serde_json::Value, jsonrpc_core::Error> as std::convert::From<ClientResponse>>
              <std::result::Result<serde_json::Value, jsonrpc_core::Error> as std::convert::From<jsonrpc_core::Output>>
    = note: required because of the requirements on the impl of `Into<std::result::Result<solana_rpc_client_api::response::Response<std::collections::HashMap<solana_rpc_client_api::config::RpcAccountIndex, usize>>, jsonrpc_core::Error>>` for `solana_rpc_client_api::custom_error::RpcCustomError` ```

Reply:
Trent's suggestion is just missing the `Err()` enum wrapper.